### ISSUE TITLE
服务端崩溃，报RpcException(Forbid consumer)时，mock无效

### ISSUE THREAD

**Issue by paopaoteng** [apache/incubator-dubbo#1498]
在xml文件里配置<dubbo:reference  id="service"    interface="com.xxx.service.publicInfo.ISystemConfigService" check="false" mock="com.xxx.service.publicInfo.SystemConfigServiceMock" />
当服务端全部挂掉时，客户端调用服务报Forbid consumer的RpcException，此时Mock没有生效。
看源码是RegistryDirectory.refreshInvoker中将forbidden修改为true，此时mock也会报同样的错误。

**Comment by apel-apel** [comment]
### similar question
I feel that the mock is not work,this is my spring xml config about dubbo below：

provider.xml
```xml 
<!-- provider's application name, used for tracing dependency relationship -->
    <dubbo:application name="demo-provider"/>

    <!-- use multicast registry center to export service -->
    <dubbo:registry address="multicast://224.5.6.7:1234"/>

    <!-- use dubbo protocol to export service on port 20880 -->
    <dubbo:protocol name="dubbo" port="20880"/>

    <!-- service implementation, as same as regular local bean -->
    <bean id="demoService" class="com.alibaba.dubbo.demo.provider.DemoServiceImpl"/>

    <!-- declare the service interface to be exported -->
    <!--<dubbo:service interface="com.alibaba.dubbo.demo.DemoService" mock="com.alibaba.dubbo.demo.DemoServiceMock"/>-->
    <dubbo:service interface="com.alibaba.dubbo.demo.DemoService" ref="demoService" mock="com.alibaba.dubbo.demo.DemoServiceMock" />
```
consumer.xml
```xml
<!-- consumer's application name, used for tracing dependency relationship (not a matching criterion),
    don't set it same as provider -->
    <dubbo:application name="demo-consumer"/>

    <!-- use multicast registry center to discover service -->
    <dubbo:registry address="multicast://224.5.6.7:1234"/>

    <!-- generate proxy for the remote service, then demoService can be used in the same way as the
    local regular interface -->
    <dubbo:reference id="demoService" check="false" interface="com.alibaba.dubbo.demo.DemoService" />
```
my provider obviously throw a RpcExcepton, and I expect the mock result return, but consumer still throws the exception;

my provider class:
```java
public class DemoServiceImpl implements DemoService {

    public String sayHello(String name) {
        System.out.println("[" + new SimpleDateFormat("HH:mm:ss").format(new Date()) + "] Hello " + name + ", request from consumer: " + RpcContext.getContext().getRemoteAddress());
        if (true) {
            throw new RpcException("system error");
        }
        return "Hello " + name + ", response form provider: " + RpcContext.getContext().getLocalAddress();
    }

}
```

my mock class:
```java
public class DemoServiceMock implements DemoService {

    public DemoServiceMock() {
    }

    @Override
    public String sayHello(String name) {
        return "I am mock";
    }
}
```

is there something wrong?

**Comment by vangoleo** [comment]
Hi, I didn't reproduce the issue by using your config. 
Another user also use the same configuration, and it does work in his side. Please refer to below link:
https://github.com/apache/incubator-dubbo/issues/1501

Could you please provide more details of your code and config? Thanks.

**Comment by chickenlj** [comment]
@paopaoteng Which dubbo version do you use?

This issue has been resolved in 2.5.6.

**Comment by chickenlj** [comment]
Please check, feel free to reopen if necessary
